examples/platforms/posix/flash.c

```diff
@@ -98,19 +99,70 @@ uint32_t utilsFlashGetSize(void)
 ThreadError utilsFlashErasePage(uint32_t aAddress)
 {
     ThreadError error = kThreadError_None;
-    uint8_t buf = 0xff;
     uint32_t address;
+    uint8_t dummy_page[ FLASH_SIZE ];
 
     otEXPECT_ACTION(sFlashFd >= 0, error = kThreadError_Failed);
     otEXPECT_ACTION(aAddress < FLASH_SIZE, error = kThreadError_InvalidArgs);
 
     // Get start address of the flash page that includes aAddress
     address = aAddress & (~(uint32_t)(FLASH_PAGE_SIZE - 1));
 
-    for (uint16_t offset = 0; offset < FLASH_PAGE_SIZE; offset++)
-    {
-        otEXPECT_ACTION(pwrite(sFlashFd, &buf, 1, address + offset) == 1, error = kThreadError_Failed);
-    }
+    // Some notes:
```

Comment:
The notes below are great!  At the same time, I think this might be better placed in the commit message rather directly in the code.  I don't think there was a good reason to use the `for-loop` implementation and I can't think of a reason why someone might want to revert back to that implementation.